File: cpp/example_code/cloudwatch/disable_alarm_actions.cpp
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX - License - Identifier: Apache - 2.0
/*

*/
/* ////////////////////////////////////////////////////////////////////////////
 * Purpose: disable_alarm_actions.cpp demonstrates how to disable actions on an Amazon CloudWatch alarm.
 *
 * Prerequisites:
 * An Amazon CloudWatch metric alarm with at least one action.
 *
 * Inputs:
 * - alarm_name (entered as second argument in command line)
 *
 * Outputs:
 * The alarm is disabled.
 * ///////////////////////////////////////////////////////////////////////// */
//snippet-start:[cw.cpp.disable_alarm_actions.inc]
#include <aws/core/Aws.h>
#include <aws/monitoring/CloudWatchClient.h>
#include <aws/monitoring/model/DisableAlarmActionsRequest.h>
#include <iostream>
//snippet-end:[cw.cpp.disable_alarm_actions.inc]

/**
 * Disable actions on a CloudWatch alarm based on command-line input
 */
int main(int argc, char** argv)
{
    if (argc != 2)
    {
        std::cout << "Usage: disable_alarm_actions <alarm_name>" <<
            std::endl;
        return 1;
    }

    Aws::SDKOptions options;

    Aws::InitAPI(options);
    {
        Aws::String alarm_name(argv[1]);

        //snippet-start:[cw.cpp.disable_alarm_actions.code]
        Aws::CloudWatch::CloudWatchClient cw;

        Aws::CloudWatch::Model::DisableAlarmActionsRequest disableAlarmActionsRequest;
        disableAlarmActionsRequest.AddAlarmNames(alarm_name);

        auto disableAlarmActionsOutcome = cw.DisableAlarmActions(disableAlarmActionsRequest);
        if (!disableAlarmActionsOutcome.IsSuccess())
        {
            std::cout << "Failed to disable actions for alarm " << alarm_name <<
                ": " << disableAlarmActionsOutcome.GetError().GetMessage() <<
                std::endl;
        }
        else
        {
            std::cout << "Successfully disabled actions for alarm " <<
                alarm_name << std::endl;
        }
        //snippet-end:[cw.cpp.disable_alarm_actions.code]
    }
    Aws::ShutdownAPI(options);
    return 0;
}

File: dotnetv3/CloudWatch/DisableAlarmActionsExample/DisableAlarmActionsExample/DisableAlarmActions.cs
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier:  Apache-2.0

namespace DisableAlarmActionsExammple
{
    // snippet-start:[CloudWatch.dotnetv3.DisableAlarmActionsExammple]
    using System;
    using System.Collections.Generic;
    using System.Threading.Tasks;
    using Amazon.CloudWatch;
    using Amazon.CloudWatch.Model;

    /// <summary>
    /// This example shows how to disable the Amazon CloudWatch actions for
    /// one or more CloudWatch alarms. The example was created using the
    /// AWS SDK for .NET version 3.7 and .NET Core 5.0.
    /// </summary>
    public class DisableAlarmActions
    {
        public static async Task Main()
        {
            IAmazonCloudWatch cwClient = new AmazonCloudWatchClient();
            var alarmNames = new List<string>
            {
                "ALARM_NAME",
                "ALARM_NAME_2",
            };

            var success = await DisableAlarmsActionsAsync(cwClient, alarmNames);

            if (success)
            {
                Console.WriteLine("Alarm action(s) successfully disabled.");
            }
            else
            {
                Console.WriteLine("Alarm action(s) were not disabled.")
            }
        }

        /// <summary>
        /// Disable the actions for the list of CloudWatch alarm names passed
        /// in the alarmNames parameter.
        /// </summary>
        /// <param name="client">An initialized CloudWatch client object.</param>
        /// <param name="alarmNames">The list of CloudWatch alarms to disable.</param>
        /// <returns>A Boolean value indicating the success of the call.</returns>
        public static async Task<bool> DisableAlarmsActionsAsync(
            IAmazonCloudWatch client,
            List<string> alarmNames)
        {
            var request = new DisableAlarmActionsRequest
            {
                AlarmNames = alarmNames,
            };

            var response = await client.DisableAlarmActionsAsync(request);

            return response.HttpStatusCode == System.Net.HttpStatusCode.OK;
        }
    }

    // snippet-end:[CloudWatch.dotnetv3.DisableAlarmActionsExammple]
}
